esphome/components/atm90e32/atm90e32.h

```diff
@@ -119,21 +188,50 @@ class ATM90E32Component : public PollingComponent,
     uint32_t cumulative_reverse_active_energy_{0};
   } phase_[3];
 
-  struct Calibration {
-    uint16_t voltage_offset_{0};
-    uint16_t current_offset_{0};
+  struct OffsetCalibration {
+    int16_t voltage_offset_{0};
+    int16_t current_offset_{0};
   } offset_phase_[3];
 
-  ESPPreferenceObject pref_;
+  struct PowerOffsetCalibration {
+    int16_t active_power_offset{0};
+    int16_t reactive_power_offset{0};
+  } power_offset_phase_[3];
+
+  struct GainCalibration {
+    uint16_t voltage_gain{1};
+    uint16_t current_gain{1};
+  } gain_phase_[3];
+
+  ESPPreferenceObject offset_pref_;
+  ESPPreferenceObject power_offset_pref_;
+  ESPPreferenceObject gain_calibration_pref_;
 
   sensor::Sensor *freq_sensor_{nullptr};
+  text_sensor::TextSensor *phase_status_text_sensor_[3]{nullptr};
+  text_sensor::TextSensor *freq_status_text_sensor_{nullptr};
```

Comment:
Any code like this that touches any other platform needs to be wrapped in `ifdefs`
This applies to any includes, references like here and code that touches those references.

```suggestion
#ifdef USE_TEXT_SENSOR
  text_sensor::TextSensor *phase_status_text_sensor_[3]{nullptr};
  text_sensor::TextSensor *freq_status_text_sensor_{nullptr};
#endif
```